### PR TITLE
[2/2] Eng 1625 Adds frontend structs and handler for getWorkflowDagResult API

### PR DESCRIPTION
**Pull request by likawind**
## Describe your changes and why you are making these changes
This PR adds frontend response structs and async thunk code to handle `getWorkflowDagResult` API introduced in #425 . It mainly mirrors response structs from golang with a simple async thunk to process the request. We need to test this together with the new UI code

## Related issue number (if any)
ENG-1625

## Checklist before requesting a review
This PR only adds stuff that's not called, so no test is necessary as long as linter passes.
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/ui/common/src/handlers/responses/dag.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  schedule?: WorkflowSchedule;
+  retention_policy?: WorkflowRetentionPolicy;
+};
+
+export type DagResponse = DagMetadataResponse & {
+  operators: { [id: string]: OperatorResponse };
+  artifacts: { [id: string]: ArtifactResponse };
+};
+
+export type DagRawResultResponse = {
```
Comment (eunice-chan) [inline src/ui/common/src/handlers/responses/dag.ts:32]: I wonder if there is a more descriptive name for this result. DagRawResultResponse doesn't let me know what it contains. I think DagMetadatResponse would be more descriptive but that name is already in use.

**Review by eunice-chan — APPROVED** [review]
I like the comments about the associated `.go` file for the `.ts` files you wrote ("this file should mirror...")!

File: src/ui/common/src/handlers/responses/dag.ts (hunk 1 above)
Comment (likawind) [reply src/ui/common/src/handlers/responses/dag.ts:32]: yeah I had a hard time figuring this out as well. This is technically a `ResultResponse` but that's also used by the main `ResultResponse`

Comment (agiron123) [reply src/ui/common/src/handlers/responses/dag.ts:32]: Maybe use something like DagResultStatus here @likawind ?

**Review by agiron123 — APPROVED** [review]
Nice work! Small comment about naming below, but after resolved you can go ahead and merge.

**Pull request merged**